tests/test_processors.py

```diff
@@ -714,3 +729,339 @@ def test_py3_exception_no_traceback(self):
+
+            assert {"event": "baz", **callsite_params} == json.loads(
+                string_io.getvalue()
+            )
+
+        finally:
+            structlog.reset_defaults()
+
+    def test_additional_ignores(self, monkeypatch: pytest.MonkeyPatch) -> None:
+        """
+        Stack frames from modules with names that start with values in
+        `additional_ignores` are ignored when determining the callsite.
+        """
+        test_message = "test message"
+        additional_ignores = ["tests.utils"]
+        processor = self.make_processor(None, additional_ignores)
+        event_dict: EventDict = {"event": test_message}
+
+        # WARNING: The below three lines are sensitive to relative line numbers
+        # (i.e. the invocation of processor must be two lines after the
+        # invocation of get_callsite_parameters) and is order sensitive (i.e.
+        # monkeypatch.setattr must occur after get_callsite_parameters but
+        # before invocation of processor).
+        callsite_params = self.get_callsite_parameters(2)
+        monkeypatch.setattr(sys, "_getframe", value=mock_getframe)
+        actual = processor(None, None, event_dict)
+
+        expected = {
+            "event": test_message,
+            **callsite_params,
+        }
+        assert expected == actual
+
+    @pytest.mark.parametrize(
+        "origin, parameter_strings",
+        itertools.product(
+            ["logging", "structlog"],
+            [
+                None,
+                *[{parameter} for parameter in parameter_strings],
+                set(),
+                parameter_strings,
+                {"pathname", "filename"},
+                {"module", "func_name"},
+            ],
+        ),
+    )
+    def test_processor(
+        self,
+        origin: str,
+        parameter_strings: Optional[Set[str]],
+    ):
+        """
+        The correct callsite parameters are added to event dictionaries.
+        """
+        test_message = "test message"
+        processor = self.make_processor(parameter_strings)
+        if origin == "structlog":
+            event_dict: EventDict = {"event": test_message}
+            callsite_params = self.get_callsite_parameters()
+            actual = processor(None, None, event_dict)
+        elif origin == "logging":
+            callsite_params = self.get_callsite_parameters()
+            record = logging.LogRecord(
+                "name",
+                logging.INFO,
+                callsite_params["pathname"],
+                callsite_params["lineno"],
+                test_message,
+                None,
+                None,
+                callsite_params["func_name"],
+            )
+            event_dict: EventDict = {
+                "event": test_message,
+                "_record": record,
+                "_from_structlog": False,
+            }
+            actual = processor(None, None, event_dict)
+        else:
+            raise ValueError(f"invalid origin {origin}")
```

Comment:
I think `pytest.fail` is the more idiomatic choice here? This applies to several other spots too, but I won't pester you with comment of them all. :)

Reply:
It is, I will change it.

Reply:
Changed `ValueError` to `pytest.fail` in all tests I added.